scripts/test_update_models.py

```
@@ -426,6 +426,61 @@ def test_none_old_model_returns_new_unchanged(self) -> None:
+    def test_endpoint_survives_ordered_model(self) -> None:
+        # ordered_model only emits fields in FIELD_ORDER; endpoint must be there
+        # so the preserved value is not dropped during serialization.
+        ordered = um.ordered_model({"name": "gpt-5", "endpoint": "responses"})
+        self.assertEqual(ordered["endpoint"], "responses")
+
+
+class TestOpenAIEndpointDefault(unittest.TestCase):
+    """OpenAI chat models are routed to the responses endpoint when the LiteLLM
+    registry's supported_endpoints signals it."""
+
+    _RESP = ["/v1/chat/completions", "/v1/batch", "/v1/responses"]
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)